### PR TITLE
More Python 3 fixes

### PR DESCRIPTION
**Pull request by hannosch**
All Python 3 versions now see the same number of test failures:

Total: 1118 tests, 28 failures, 33 errors and 1 skipped in 7.121 seconds.

### CODE REVIEW

File: src/Products/Five/browser/tests/resource.txt [hunk 1]
```diff
@@ -65,7 +65,7 @@ PageTemplateResource's __call__ renders the template
 
   >>> for r in dir_resource_names:
   ...     resource = self.folder.unrestrictedTraverse(base % r)
-  ...     self.assert_(isinstance(resource, Resource))
+  ...     self.assertTrue(isinstance(resource, Resource))
```
Comment (tseaver) [inline src/Products/Five/browser/tests/resource.txt]: Could use `self.assertIsInstance(resource, Resource)`.

File: src/Products/PageTemplates/tests/secure.pt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <div xmlns="http://www.w3.org/1999/xhtml"
      xmlns:tal="http://xml.zope.org/namespaces/tal">
   <span tal:define="soup view/tagsoup | options/soup"
-        tal:replace="structure python: modules['cgi'].escape(soup)" />
+        tal:replace="structure python: modules['cgi'].escape(soup, True)" />
```
Comment (tseaver) [inline src/Products/PageTemplates/tests/secure.pt]: Shouldn't we be avoiding `cgi.escape`?

File: src/Testing/ZopeTestCase/zopedoctest/testFunctionalDocTest.py [hunk 3]
```diff
@@ -89,23 +89,24 @@ def test___str___w_headers(self):
 
 
 SHOW_COOKIES_DTML = '''\
-<dtml-in "REQUEST.cookies.keys()">
+<dtml-in "list(REQUEST.cookies.keys())">
```
Comment (tseaver) [inline src/Testing/ZopeTestCase/zopedoctest/testFunctionalDocTest.py]: Just iterating doesn't work under Py3k?

File: src/Products/PageTemplates/tests/secure.pt (hunk 2 above)
Comment (hannosch) [inline src/Products/PageTemplates/tests/secure.pt]: We should. But it's still there in Python 3.6, so it's not required for porting.

I couldn't remember how to write a conditional import or the equivalent thereof inside a restricted page template. IIRC someone has to declare the cgi (or html) module and the escape function as safe to use as well.

File: src/Testing/ZopeTestCase/zopedoctest/testFunctionalDocTest.py (hunk 3 above)
Comment (hannosch) [inline src/Testing/ZopeTestCase/zopedoctest/testFunctionalDocTest.py]: This is admittedly me papering over an issue, so I can see the other errors.

This failed with something like SafeIter has no attribute `__len__` and that's most likely an AccessControl issue.

File: src/Products/Five/browser/tests/resource.txt (hunk 1 above)
Comment (hannosch) [inline src/Products/Five/browser/tests/resource.txt]: Ah, I didn't even know about that one.

**Pull request merged**